### ISSUE TITLE
Infinite loop in inline_line_width

### ISSUE THREAD

**Issue by p-vitt** [Kozea/WeasyPrint#557]
I'm constantly running into an infinte loop in [inline_line_width](https://github.com/Kozea/WeasyPrint/blob/master/weasyprint/layout/preferred.py#L252) when trying to dump a particular website. With others, it runs fine. 

Some system details:

```
$ pip list --format=columns | grep WeasyPrint
WeasyPrint        0.42
$ python --version
Python 3.6.4
$ uname -mrs
Darwin 15.6.0 x86_64
```

`new_resume_at` stays to be 0 and `resume_at` stays to be 55 in every iteration, thus this loop won't end. At the beginning of the loop, some values get added to `lines`, but later on it keeps adding 0.9431584 on and on again.

How to reproduce:

```
link = 'http://www.zeit.de/politik/deutschland/2018-01/gruene-robert-habeck-palastrevolte-fluegelstreit-amt-und-mandat/komplettansicht'
weasyprint.HTML(url=link).write_pdf("test.pdf")
```

Do I misuse weasyprint? Do I need to add some template or configuration in order to dump a website?

**Comment by liZe** [comment]
> Do I misuse weasyprint?

No, you don't. There's a bug!

**Comment by liZe** [comment]
I've tried hard to build a small test case to reproduce the problem and add it to WP's tests, but the real case is so strange (`::after` pseudo-tag in a `display: table` `a` tag with `content: attr(href)` and `overflow-wrap: break-word`) that I failed.

By the way, bug is fixed in 995a9a3, the cause was easy to spot and the fix is clean.
